src/components/MenuItem/MenuItem.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import classnames from "classnames";
+import React, {
+  ComponentPropsWithoutRef,
+  ComponentType,
+  ElementType,
+  SVGAttributes,
+} from "react";
+import styles from "./MenuItem.module.css";
+import { Text } from "../Typography/Text";
+import ChevronRightIcon from "@vector-im/compound-design-tokens/icons/chevron-right.svg";
+
+type MenuItemElement = "button" | "label" | "div";
+
+type Props<C extends MenuItemElement> = {
+  /**
+   * The element type of this menu item.
+   * @default button
+   */
+  as?: C;
+  className?: string;
+  /**
+   * The icon to show on this menu item.
+   */
+  Icon: ComponentType<SVGAttributes<SVGElement>>;
+  /**
+   * The label to show on this menu item.
+   */
+  label: string;
+  /**
+   * The color variant of the menu item.
+   * @default primary
+   */
+  kind?: "primary" | "critical";
+  /**
+   * Whether to replace the children with a navigation hint on hover.
+   * @default true
+   */
+  navHint?: boolean;
```

Comment:
I'm unsure about that naming. It appears fine, but I don't understand what conditions warrant the chevron to be hidden.
I've raised that question in an internal room, and hopefully @americanrefugee might be able to add extra context and we can settle on something final here.

Reply:
The use case is when you have `<input>`s as the child, for example in the [toggle menu item](https://github.com/vector-im/compound-web/pull/92) scenario. These shouldn't be hidden on hover, because that would make it awkward to interact with them, and the purpose of the menu item isn't for navigation.

Reply:
right, so i believe we could infer the visibility of the chevron based on whether the component has an `href` or `onClick` handler. no?

Reply:
sure, makes sense to me